tests/study_tests/test_study.py

```diff
@@ -1485,22 +1485,12 @@ def test_tell_duplicate_tell() -> None:
 def test_tell_storage_not_implemented_trial_number() -> None:
     with StorageSupplier("inmemory") as storage:
 
-        with patch.object(
-            storage,
-            "get_trial_id_from_study_id_trial_number",
-            side_effect=NotImplementedError,
-        ):
-            study = create_study(storage=storage)
-
-            study.tell(study.ask(), 1.0)
-
-            # Storage missing implementation for method required to map trial numbers back to
-            # trial IDs.
-            with pytest.warns(UserWarning):
-                study.tell(study.ask().number, 1.0)
-
-            with pytest.raises(ValueError):
-                study.tell(study.ask().number + 1, 1.0)
+        study = create_study(storage=storage)
+        study.tell(study.ask(), 1.0)
+        study.tell(study.ask().number, 1.0)
+
+        with pytest.raises(ValueError):
+            study.tell(study.ask().number + 1, 1.0)
```

Comment:
With this, `InMemoryStorage.get_trial_id_from_study_id_trial_number` is tested instead of `BaseStorage.get_trial_id_from_study_id_trial_number`.

Reply:
I cannot think of a way to easily test `InMemoryStorage.get_trial_id_from_study_id_trial_number` (except for mocking all the abstract methods of `InMemoryStorage`), but do you come up with a good way...?

Reply:
Good catch! It is a difficult problem how to test concrete methods in an abstract base class. I thought about it and decided it would be better to remove this test case. At least it should not be in `test_study.py`.

Reply:
It seems that `get_trial_id_from_study_id_trial_number()` is enough tested at:

https://github.com/optuna/optuna/blob/2c09c5d689411aae19a613970ade3a1b54b01ddb/tests/storages_tests/test_storages.py#L1070-L1096

Reply:
I agree that this test should not be located in `test_study.py`.
Still, I feel the default logic of `BaseStorage.get_trial_id_from_study_id_trial_number` should be tested somewhere (not tested in `test_get_trial_id_from_study_id_trial_number` because all the concrete class override the method). We should probably discuss how to test default implementations in `BaseStorage` as another issue.

Reply:
So, let me merge this PR first. If you have any opinions, please leave a comment!